Fix AthenaSQLHook to handle hook_params properly

sidshas03 opened the pull request:
- Extract Athena-specific parameters from kwargs before passing to AwsBaseHook
- This fixes TypeError when using SQLValueCheckOperator with AthenaSQLHook
- Parameters like s3_staging_dir and work_group are now properly handled
- Maintains backward compatibility and follows pattern used by other SQL hooks

Fixes #55678

sidshas03 committed:
Fix AthenaSQLHook to handle hook_params properly

- Extract Athena-specific parameters from kwargs before passing to AwsBaseHook
- This fixes TypeError when using SQLValueCheckOperator with AthenaSQLHook
- Parameters like s3_staging_dir and work_group are now properly handled
- Maintains backward compatibility and follows pattern used by other SQL hooks

Fixes #55678

sidshas03 committed:
Merge branch 'main' into fix/athena-sql-hook-params-55678-clean

sidshas03 committed:
Merge branch 'main' into fix/athena-sql-hook-params-55678-clean

sidshas03 committed:
Merge branch 'main' into fix/athena-sql-hook-params-55678-clean

vincbeck commented on providers/amazon/src/airflow/providers/amazon/aws/hooks/athena_sql.py:
```diff
@@ -66,6 +66,20 @@ class AthenaSQLHook(AwsBaseHook, DbApiHook):
+        self.assume_role_kwargs = kwargs.pop("assume_role_kwargs", None)
+        self.aws_session_token = kwargs.pop("aws_session_token", None)
+        self.endpoint_url = kwargs.pop("endpoint_url", None)
+        
```
Instead of doing that, should we not declare them in the constructor as optional parameter. Example: `s3_staging_dir: str | None = None`?

o-nikolas replied:
That would make the code inside the __init__ body cleaner for sure. But then it would make the signature a bit more complicated which is more user facing? If these args are used frequently then they should be in the signature, but if not maybe it's best this way?

jroachgolf84 replied:
I agree with @vincbeck, this matches the pattern for other AWS Operators.

sidshas03 replied:
Thanks for the input, @vincbeck @o-nikolas @jroachgolf84 — I’ve applied the constructor change in dbe991662c.

What changed:
- Switched AthenaSQLHook to explicit __init__ params (s3_staging_dir, work_group, driver, aws_domain, session_kwargs, config_kwargs, role_arn, assume_role_method, assume_role_kwargs, aws_session_token, endpoint_url) and kept **kwargs for compatibility.
- Extract athena_conn_id before calling super().__init__(...).
- Behaviour unchanged: get_conn() still honours driver/aws_domain from self → hook params > extras.
- Breeze static checks + targeted Amazon Non-DB tests pass.

Could you please confirm the param order/names look good and if you’d like any tweaks to the class/__init__ docstring? Happy to adjust.

o-nikolas commented on providers/amazon/tests/unit/amazon/aws/hooks/test_athena_sql.py:
```diff
@@ -153,3 +153,91 @@ def test_conn_id_override_setter(self):
+        from airflow.providers.common.sql.operators.sql import SQLValueCheckOperator
+        from unittest.mock import patch
+        
+        # Mock Athena connection with s3_staging_dir in extra (as described in the issue)
```
"The issue" is context for you/AI at the time of writing the code. In the future it will not have meaning, we can't keep references to every issue/PR in our code itself. Don't leave references to it in the code permanently.

```suggestion
        # Mock Athena connection with s3_staging_dir in extra
```

o-nikolas commented on providers/amazon/tests/unit/amazon/aws/hooks/test_athena_sql.py:
```diff
@@ -153,3 +153,91 @@ def test_conn_id_override_setter(self):
+        assert call_args["work_group"] == "test-workgroup"
+
+    def test_sql_value_check_operator_compatibility(self):
+        """Test that AthenaSQLHook works with SQLValueCheckOperator (reproduces issue #55678)."""
```
See below.
```suggestion 
        """Test that AthenaSQLHook works with SQLValueCheckOperator"""
```

sidshas03 committed:
Merge branch 'main' into fix/athena-sql-hook-params-55678-clean

sidshas03 committed:
AthenaSQLHook: refine get_conn() and tests (no functional change)

sidshas03 committed:
AthenaSQLHook: switch to explicit __init__ params (replacing kwargs.pop pattern)

- Added explicit optional params for s3_staging_dir, work_group, driver, aws_domain, session_kwargs, config_kwargs, role_arn, assume_role_method, assume_role_kwargs, aws_session_token, endpoint_url
- Kept **kwargs for backward compatibility
- Extract athena_conn_id from kwargs before calling super().__init__(...)
- get_conn() continues to honor driver/aws_domain from self (hook params > extras)
- Tests: test_hook_params_override_extras() still passes; imports/patching cleaned; issue refs removed
- Aligns with pattern used by other AWS operators
- Fully backward-compatible

vincbeck commented on providers/amazon/src/airflow/providers/amazon/aws/hooks/athena_sql.py:
```diff
@@ -65,8 +65,41 @@ class AthenaSQLHook(AwsBaseHook, DbApiHook):
-        super().__init__(*args, **kwargs)
+    def __init__(
+        self,
+        athena_conn_id: str | None = None,  # keep positional compatibility
```
Keep it as is?

```suggestion
        athena_conn_id: str = default_conn_name
```

sidshas03 replied:
Thanks! I’ll change the signature to
This keeps positional/keyword compatibility without extra fallback code. I’ll run `prek` and push.
> `athena_conn_id: str = default_conn_name`.

vincbeck commented on providers/amazon/src/airflow/providers/amazon/aws/hooks/athena_sql.py:
```diff
@@ -65,8 +65,41 @@ class AthenaSQLHook(AwsBaseHook, DbApiHook):
+        if athena_conn_id is None:
+            athena_conn_id = kwargs.pop("athena_conn_id", self.default_conn_name)
+        else:
+            kwargs.pop("athena_conn_id", None)  # avoid conflicts
```
I do not think you need that

sidshas03 replied:
Agreed. With the default in the signature, the `kwargs.pop("athena_conn_id", …)` block isn’t needed. I’ll remove lines 85–89 and set `self.athena_conn_id = athena_conn_id` directly. No behavior change elsewhere.

o-nikolas commented on providers/amazon/src/airflow/providers/amazon/aws/hooks/athena_sql.py:
```diff
@@ -65,8 +65,41 @@ class AthenaSQLHook(AwsBaseHook, DbApiHook):
+        else:
+            kwargs.pop("athena_conn_id", None)  # avoid conflicts
+        super().__init__(**kwargs)
+        # Store explicit params on self
```
These types of comments (that often come from AI) aren't very useful in the final code.

o-nikolas replied:
Btw, why did we flip back to setting all of these on the init constructor vs fetching them from kwargs? Didn't you say in a previous message that you were going to keep the kwargs pop?

sidshas03 replied:
> Btw, why did we flip back to setting all of these on the init constructor vs fetching them from kwargs? Didn't you say in a previous message that you were going to keep the kwargs pop?

Right, I moved to explicit __init__ params based on the feedback from @vincbeck and @jroachgolf84 to match other AWS hooks. I’ve removed the kwargs.pop fallback entirely and kept athena_conn_id: str = default_conn_name for BC. No functional change otherwise.

o-nikolas commented on providers/amazon/tests/unit/amazon/aws/hooks/test_athena_sql.py:
```diff
@@ -153,3 +163,123 @@ def test_conn_id_override_setter(self):
+
+    def test_hook_params_handling(self):
+        """Test that hook_params are properly handled and don't cause TypeError."""
+        # Test that hook_params with Athena-specific parameters don't cause errors
```
You don't need both of these

sidshas03 replied:
Agreed! I’ll keep the docstring and remove the redundant inline comment.

o-nikolas commented on providers/amazon/tests/unit/amazon/aws/hooks/test_athena_sql.py:
```diff
@@ -153,3 +163,123 @@ def test_conn_id_override_setter(self):
+        hook.get_connection = mock.Mock(return_value=conn)
+
+        # Call get_conn
+        hook.get_conn()
```
I love comments just as much as the next person (if not more even), but these kinds of comments are really unnecessary.

sidshas03 replied:
Acknowledged! I’ll delete that line and similar noise. Only keeping comments that explain non-obvious behavior.

o-nikolas replied:
I still think some that are left are not helpful. But I'm not going to burn more time on this PR.

o-nikolas commented on providers/amazon/tests/unit/amazon/aws/hooks/test_athena_sql.py:
```diff
@@ -153,3 +163,123 @@ def test_conn_id_override_setter(self):
+        )
+
+        with patch(f"{BASEHOOK_PATCH_PATH}.get_connection", return_value=athena_conn):
+            # This should NOT raise TypeError: AwsGenericHook.__init__() got an unexpected keyword argument 's3_staging_dir'
```
This is a great comment 🙂

sidshas03 replied:
Thanks! I’ll keep this one as is since it clarifies the intent of the patch block.

sidshas03 committed:
AthenaSQLHook: put default_conn_name before __init__; drop athena_conn_id fallback; remove filler comments (no functional change)

sidshas03 committed:
chore(AthenaSQLHook/tests): remove filler comments (no functional change)
